src/BaseFactory.php

```diff
@@ -46,16 +45,29 @@ protected function build(array $extra = [], string $creationType = 'create')
             return $model;
         }
 
-        $relatedModels = $this->relatedModelFactories->map->make();
-
-        if ($creationType === 'create') {
-            $model->{$this->relatedModelRelationshipName}()
-                    ->saveMany($relatedModels);
-
-            return $model;
+        foreach ($this->relatedModelFactories as $relationshipName => $factories) {
```

Comment:
As you said we should put this part regarding the relations somewhere else. Not sure If I want a separate class to handle this. Let's put it to a new method in the BaseFactory for now.

Reply:
I created a `buildRelationsForModel` method to build model's relations. It's private as it's not intendeed to be called from userland factories.